kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -191,6 +197,42 @@ private void handleMetadata(ColumnVector metadataVector) {
     }
   }
 
+  /**
+   * Checks whether each of the current transaction's {@link DomainMetadata} conflicts with the
+   * winning transaction at any domain.
+   *
+   * <ol>
+   *   <li>Accept the current transaction if its set of metadata domains does not overlap with the
+   *       winning transaction's set of metadata domains.
+   *   <li>Otherwise, fail the current transaction unless each conflicting domain is associated with
+   *       a domain-specific way of resolving the conflict.
+   * </ol>
+   *
+   * @param domainMetadataVector domainMetadata rows from the winning transactions
+   */
+  private void handleDomainMetadata(ColumnVector domainMetadataVector) {
+    // Extract the domain metadata map from the winning transaction.
+    Map<String, DomainMetadata> winningDomainMetadataMap =
+        DomainMetadataUtils.extractDomainMetadataMap(domainMetadataVector);
+
+    for (DomainMetadata dm : this.transaction.getDomainMetadatas()) {
+      // Try to resolve the conflict, if not possible, throw a ConcurrentTransaction exception
+      resolveDomainMetadataConflict(dm, winningDomainMetadataMap);
```

Comment:
nit: you can delete the function `resolveDomainMetadataConflict` and just put the logic here ... the funciton is simple enough and isn't used anywhere else